thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java

```diff
@@ -994,7 +995,10 @@ private String getDimensions(String dataset, String exploreDimensions) throws Ex
       }
 
       for (MergedAnomalyResultDTO dto : mergedResults) {
-        idList.add(dto.getId());
+        // if use notified flag, only keep anomalies isNotified == true
+        if ( (useNotified && dto.isNotified()) || !useNotified) {
```

Comment:
Please rename dto to mergedAnomaly and idList to anomalyIdList. Improve the naming in other places too.

Reply:
fixed. thanks